src/expressions/operator-expr.md

```diff
@@ -396,6 +396,11 @@ Casts an enum to its discriminant, then uses a numeric cast if needed.
 
 Casts to the `char` with the corresponding code point.
 
+#### Pointer to pointer cast
+
+Casting from a pointer to unsized type to pointer to sized type discards the pointer metadata, resulting in just a pointer to the referenced memory.
+Casting between pointers to unsized type preserves the pointer metadata unchanged (e.g. the slice element length remains the same).
```

Comment:
Could this document an example of how this can be unexpected? For instance, an `as` from `*const [u8]` to `*const [u32]` will end up pointing to four times as much memory?

Reply:
Here's a potential illustration that seems appropriate for the reference:

> To illustrate:
>
> ```rust
> let u8_slice_ptr = ptr::slice_from_raw_parts::<u8>(ptr::null(), 4);
> assert_eq!(4, mem::size_of_val_raw(u8_slice_ptr));
> let u8_ptr = u8_slice_ptr as *const u8;
> assert_eq!(1, mem::size_of_val_raw(u8_ptr));
> let u16_slice_ptr = u8_slice_ptr as *const [u16];
> assert_eq!(8, mem::size_of_val_raw(u16_slice_ptr));
> ```

Alternatively:

> To illustrate:
> 
> ```rust
> let u8_slice_ptr = ptr::slice_from_raw_parts::<u8>(ptr::null(), 4);
> assert_eq!((ptr::null(), 4), u8_slice_ptr.to_raw_parts());
> 
> let u8_ptr = u8_slice_ptr as *const u8;
> assert_eq!((ptr::null(), ()), u8_ptr.to_raw_parts());
> 
> let u16_slice_ptr = u8_slice_ptr as *const [u16];
> assert_eq!((ptr::null(), 4), u16_slice_ptr.to_raw_parts());
> ```